gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java

```diff
@@ -275,6 +276,27 @@ public static JobDataMap updatePropsInJobDataMap(JobDataMap jobDataMap,
     return jobDataMap;
   }
 
+  public static JobDataMap deepCopyJobDataMap(JobDataMap original) {
+    JobDataMap copy = new JobDataMap();
+    for (Map.Entry<String, Object> entry : original.entrySet()) {
+      // Assuming all values stored in the JobDataMap are serializable
+      Object value = entry.getValue();
+      if(value instanceof Properties) {
+        value = deepCopyProperties((Properties) value);
+      }
+      copy.put(entry.getKey(), value);
```

Comment:
is it the true deep copy? we are not creating new keys and values and using the same one.

Reply:
I tried to make deep copy with `new String()` for every key and value but it does not pass the `Assert.assertNotSame(originalJobDataMap.get("key"), newJobDataMap.get("key"));`. All the of the values are copied over unchanged. Only one new key (`isReminder`) is added to the new `jobDataMap` and new `Properties` obj created. As long as the objects themselves are different, the original job schedules will not be affected by a new key in the reminder object.